### PR TITLE
ScholarshipAmount bug fixes

### LINKED ISSUE #719: ScholarshipAmount bugs
While editing a scholarship, noticed the following. 

1. stored values (intended for backend) show when editing a scholarship. E.g. when editing a scholarship that had an amount type set to `Full Tuition`, and changing it to `varies` shows. 

Screen Shot 2021-11-02 at 8 41 22 PM

2. amount values from a previous selection carry on to new selection. E.g. when first choosing the `Fixed` type amount for a scholarship and then changing it to `Varies`, min and max are set to fixed amount you inputted at first.
Screen Shot 2021-11-02 at 8 41 48 PM

3. `toString` shows wrong text when adding a new scholarship and only setting the min. 
Screen Shot 2021-11-02 at 8 42 28 PM

### PR DESCRIPTION
**Pull request by edwingl10**
## Motivation and Context

Fixes #719.
Adds a `fromStorage` method. Similar to the `toStorage` method we already have.
Resets `min` and `max` when the scholarship amount `type` is changed. 

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] User visible change (users will notice UI or functional changes)

## Previewing Changes

1. Go to https://s4us-pr-720.onrender.com/scholarships.
2. Edit a scholarship and you should not see values intended for backend
3. change the amount types and notice the min and max resetting

## Checklist:

- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: src/components/ScholarshipAmountField.jsx [hunk 1]
```diff
@@ -79,7 +79,11 @@ function ScholarshipAmountField(props) {
             className={classes.amountSelect}
             variant="outlined"
             value={amountType}
-            onChange={formik.handleChange}>
+            onChange={(e) => {
+              formik.setFieldValue('amount.min', 0);
+              formik.setFieldValue('amount.max', 0);
+              formik.handleChange('amount.type')(e);
```
Comment (edwingl10) [inline src/components/ScholarshipAmountField.jsx:85]: got this from [here.](https://stackoverflow.com/questions/61384089/how-to-call-two-function-in-onchange-using-formik)

File: src/types/ScholarshipAmount.ts [hunk 2]
```diff
@@ -108,6 +108,24 @@ namespace ScholarshipAmount {
     return { min, max, type };
   }
 
+  // Translates 'amount' to the format it should be displayed as
```
Comment (josueetcom) [inline src/types/ScholarshipAmount.ts:111]: Use JSDoc style as in the previous function (that way IDEs will display this comment when you hover over the `fromStorage` method).

File: src/types/ScholarshipAmount.ts [hunk 3]
```diff
@@ -117,7 +135,7 @@ namespace ScholarshipAmount {
         return `$${amount.min}`;
       case AmountType.Varies:
         if (!amount.min && !amount.max) return 'Varies';
-        if (amount.min && amount.max !== RANGE_MAX) {
+        if (amount.min && amount.max && amount.max !== RANGE_MAX) {
```
Comment (josueetcom) [inline src/types/ScholarshipAmount.ts:138]: Can you add a test case for this?

File: src/types/ScholarshipAmount.ts [hunk 4]
```diff
@@ -108,6 +108,24 @@ namespace ScholarshipAmount {
     return { min, max, type };
   }
 
+  // Translates 'amount' to the format it should be displayed as
+  export function fromStorage(amount: ScholarshipAmount): ScholarshipAmount {
```
Comment (josueetcom) [inline src/types/ScholarshipAmount.ts:112]: I don't exactly recall off the top of my head why I decided not to include this method when I added the other one. I'm guessing it has to do something with the fact that storing it again with the weird values should be ok and `toString` should be able to handle the weird values. Though I guess certain elements need the raw values and the UI would need to sort of do the same kind of handling this method does.

Comment (josueetcom) [reply src/types/ScholarshipAmount.ts:112]: We should add test cases for `toStorage` and this one 🤔.

File: src/types/ScholarshipAmount.test.ts [hunk 5]
```diff
@@ -134,3 +150,110 @@ test('toString() - Range - unbounded', () => {
...
+  expect(res).toEqual({
+    type: AmountType.Unknown,
+    min: UNKNOWN_MIN,
+    max: UNKNOWN_MAX,
+  });
+});
+
+test('fromStorage() - Fixed', () => {
+  const amount = fixed(1000);
+  const res = fromStorage(amount);
```
Comment (josueetcom) [inline src/types/ScholarshipAmount.test.ts:217]: For the `fromStorage()` tests you might want to do something like this instead to make sure you're converting from the storage represenation:
```suggestion
  const res = fromStorage(toStorage(amount));
```

**Review by josueetcom — DISMISSED** [review]

**Review by josueetcom — APPROVED** [review]

**Pull request merged**